connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -152,3 +165,208 @@ export async function getNotionAccessTokenActivity(
 
   return notionAccessToken;
 }
+
+export async function shouldGarbageCollectActivity(
```

Comment:
cc @spolu  this is the function where we decide if a given DS should go into "garbage collection" mode.
Maybe I should add something to make sure it happens during the night (but we don't have user timezones yet :/) ? For a big workspace, the process halts synchronization for ~25mn

Reply:
That seems fine for now :+1: